js/ui/grid_core/ui.grid_core.columns_controller.js

```diff
@@ -1,2634 +1,21 @@
-import $ from '../../core/renderer';
-import Callbacks from '../../core/utils/callbacks';
-import variableWrapper from '../../core/utils/variable_wrapper';
-import { compileGetter, compileSetter } from '../../core/utils/data';
-import { isDefined, isString, isNumeric, isFunction, isObject, isPlainObject, type } from '../../core/utils/type';
-import { each, map } from '../../core/utils/iterator';
-import { getDefaultAlignment } from '../../core/utils/position';
-import { extend } from '../../core/utils/extend';
-import { normalizeIndexes } from '../../core/utils/array';
-import config from '../../core/config';
-import { orderEach, deepExtendArraySafe } from '../../core/utils/object';
-import errors from '../widget/ui.errors';
-import modules from './ui.grid_core.modules';
-import gridCoreUtils from './ui.grid_core.utils';
-import { captionize } from '../../core/utils/inflector';
-import dateSerialization from '../../core/utils/date_serialization';
-import numberLocalization from '../../localization/number';
-import dateLocalization from '../../localization/date';
-import messageLocalization from '../../localization/message';
-import { when, Deferred } from '../../core/utils/deferred';
-import Store from '../../data/abstract_store';
-import { DataSource } from '../../data/data_source/data_source';
-import { normalizeDataSourceOptions } from '../../data/data_source/utils';
-import { equalByValue } from '../../core/utils/common';
-import filterUtils from '../shared/filtering';
-
-const USER_STATE_FIELD_NAMES_15_1 = ['filterValues', 'filterType', 'fixed', 'fixedPosition'];
-const USER_STATE_FIELD_NAMES = ['visibleIndex', 'dataField', 'name', 'dataType', 'width', 'visible', 'sortOrder', 'lastSortOrder', 'sortIndex', 'groupIndex', 'filterValue', 'bufferedFilterValue', 'selectedFilterOperation', 'bufferedSelectedFilterOperation', 'added'].concat(USER_STATE_FIELD_NAMES_15_1);
-const IGNORE_COLUMN_OPTION_NAMES = { visibleWidth: true, bestFitWidth: true, bufferedFilterValue: true };
-const COMMAND_EXPAND_CLASS = 'dx-command-expand';
-const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 9007199254740991/* IE11 */;
-const GROUP_COMMAND_COLUMN_NAME = 'groupExpand';
-
-const regExp = /columns\[(\d+)\]\.?/gi;
-
-let globalColumnId = 1;
+/** @deprecated */
```

Comment:
Let's change the js-doc comment to the usual comment
```javascript
/** @deprecated */
```
to
```javascript
// deprected
```

Because js-doc comments can cause some strange behaviour in our metadata-tools :(